src/util/serialise/index.js

```diff
@@ -28,29 +28,28 @@ export async function serialise (model, obj = {}, method = 'POST') {
 
     // Attributes and relationships
     for (let prop in obj) {
-      // Check if its a relationship
+      // Check if it's a relationship
       if (
         obj[prop] !== null &&
-        obj[prop].constructor === Object && (
-          typeof obj[prop].id === 'string' ||
-          typeof obj[prop].type === 'string'
-        )
+        obj[prop].constructor === Object
       ) {
-        if (typeof data.relationships === 'undefined') data.relationships = {}
-        // Guess relationship type if not provided
-        if (typeof obj[prop].type === 'undefined') obj[prop].type = this.plural(this.camel(prop))
-        data.relationships[prop] = { data: Object.assign(obj[prop]) }
+        if (typeof obj[prop].id === 'string') {
+          if (typeof data.relationships === 'undefined') data.relationships = {}
+          // Guess relationship type if not provided
+          if (typeof obj[prop].type === 'undefined') obj[prop].type = this.plural(this.camel(prop))
+          data.relationships[prop] = { data: Object.assign(obj[prop]) }
+        } else {
+          throw new Error(`${method} requires an ID for the ${prop} relationships`)
+        }
+      // Check if it's a relationship array
       } else if (
         obj[prop] !== null &&
         obj[prop].constructor === Array
       ) {
         // validate whole array
         const ptype = this.plural(this.camel(prop))
         for (var i = 0; i < obj[prop].length; i++) {
-          if (
-            typeof obj[prop][i].id === 'string' ||
-            typeof obj[prop][i].type === 'string'
-          ) {
+          if (typeof obj[prop][i].id === 'string') {
             // Guess relationship type if not provided
             if (typeof obj[prop][i].type === 'undefined') obj[prop][i].type = ptype
           } else {
```

Comment:
[Merging](https://github.com/wopian/kitsu/pull/85/files/a6dd34c2c1283c04aa157fb74ffa095aa6fd1966#r159580574) the `typeof id` and `typeof type` conditions will work if we check `id` is undefined before throwing the missing ID error.

```js
else if (typeof obj[prop][i].id === 'undefined')
```